packages/website/pages/faq.js

```diff
@@ -1,13 +1,102 @@
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
 
 import FAQPageData from '../content/pages/faq.json';
 import BlockBuilder from '../components/blockbuilder/blockbuilder.js';
 import { initFloaterAnimations } from '../lib/floater-animations.js';
 import GeneralPageData from '../content/pages/general.json';
+import AccordionBlock from '../components/accordionblock/accordionblock';
+import Loading from '../components/loading/loading';
 
 export default function Home() {
+  const [faqSections, setFaqSections] = useState();
+  /** @type [any, null | any] */
+  const [propsToParseAsMarkdown, setPropsToParseAsMarkdown] = useState([]);
   const sections = FAQPageData.page_content;
   const animations = FAQPageData.floater_animations;
+  useEffect(() => {
+    fetch(`${process.env.NEXT_PUBLIC_NETLIFY_CMS_ENDPOINT || 'https://blog.web3.storage'}/api/partials/faq/faq`)
+      .then(async response => await response.text())
+      .then(text => {
+        const obj = JSON.parse(text);
+        const faqs = obj.props.partial.meta['faq-list'].reduce((sections, faq) => {
+          sections.push({
+            id:
+              'faq-' +
+              faq.question
+                .toLowerCase()
+                .replace(/[^\w ]+/g, '')
+                .replace(/ +/g, '-'),
+            heading: faq.question,
+            content: faq.answer,
+          });
+
+          return sections;
+        }, []);
+        setFaqSections(faqs);
+        setPropsToParseAsMarkdown(['content']);
+      })
+      .catch(e => {
+        // @ts-ignore
```

Comment:
Why swallow the error here? Can you only swallow certain expected errors, and throw if you encounter an unexpected error?

Can you avoid `@ts-ignore`?

Reply:
updated to log the error and also updated to create a type for FaqSection to remove the ts-ignore.